services/ymax-planner/src/engine.ts

```diff
@@ -290,17 +279,22 @@ export const processPortfolioEvents = async (
   };
   const startFlow = async (
     portfolioStatus: StatusFor['portfolio'],
-    portfolioKey: string,
-    flowKey: string,
+    portfolioKey: PortfolioKey,
+    flowKey: FlowKey,
     flowDetail: FlowDetail,
   ) => {
     const path = `${portfoliosPathPrefix}.${portfolioKey}`;
+    const portfolioId = portfolioIdFromKey(portfolioKey);
+    const flowId = flowIdFromKey(flowKey);
+    const scope: [number, number] = [portfolioId, flowId];
+    const { policyVersion, rebalanceCount, targetAllocation } = portfolioStatus;
+    const conditions: [number, number] = [policyVersion, rebalanceCount];
```

Comment:
For types within a function, I usually prefer `as const` to provide the narrowest types, and let type inference sort the rest out.  Not a blocker if you're busy, but I'm curious about the advantages of having explicit types on `scope` and `conditions`?

```suggestion
    const scope = [portfolioId, flowId] as const;
    const { policyVersion, rebalanceCount, targetAllocation } = portfolioStatus;
    const conditions = [policyVersion, rebalanceCount] as const;
```